### DISCUSSION TITLE
Platform Terminology - what's in a name?

### DISCUSSION THREAD

**Discussion by muncus** [GoogleCloudPlatform/reliable-app-platforms#66] [Ideas]
The platform has a bunch of components with different roles. I think we could stand to use some more specific terms with them. I'm proposing some terms below, but i'd love to hear more ideas of where we need names to help associate (or separate) different components.

AIUI, there are basically 2 personas involved in RAP - the "Platform Engineer" and the "Application Engineer". They have the following rough areas of responsibility:
- App Eng: Owns one or more applications that run on the platform. responsible for choosing app archetype
- Plat Eng: Owns the infrastructure on which Apps are run, and the "on-boarding" of new Apps, and providing ready-to-use archetypes (e.g. as Terraform modules).

With that in mind, I see 2 (maybe 3) types of Management Plane here:
- App Management Plane - anything needed to build and deploy applications, like the app repo and configs like "where/how to run this"
- Platform Management Plane - this is basically the deployment archetypes. changes to these may effect many apps.
- Infrastructure Management Plane - (possibly part of the above) - this is the shared infrastructure belonging to multiple apps. Stuff like Clusters (which apps choose to use) and load balancers.

Once we have some terms we like, we can begin labeling specific bits of TF as belonging to each of these, to more clearly capture their exact purpose. WDYT?

**Comment by stevemcghee** [comment]
I think we could clarify this line of thinking by considering an even bigger potential platform, with many layers.  Does there need to be a single control, single admin plane?  of course not.    it's all relative :)

for the purposes of this project, let's consider just the perspective of the "dev" -- the app developer.  what is the admin/control plane from their perspective?  of course the platform eng will be setting intent "lower" in the stack (eg within "control" from the dev's perspective).

dev admin plane:  
- CI/CD knobs, which template to choose on onboarding (currently just 1 choice). this includes "which archetype" but does not include "what bits define an archetype"
 
dev control plane:  
- gitops pipelines - progressing new configs, argv, binaries along the path.  path determined by archetype chosen, possibly later knobs like "canary exposure time" to be added.
- cloud build, cloud deploy runs.  based on archetype chosen.
- k8s autoscaling, etc. running within the bounds of the intent expressed above. 

dev data plane:
- running clusters of nodes, pods, etc.
- future: serverless services
- all visible (via o11y), but not directly controllable by the dev.
